packages/vkui/src/components/Search/Search.tsx

```diff
@@ -111,6 +133,35 @@ export const Search = ({
     [inputRef, onCancel],
   );
 
+  const recalculateComponentStyles = React.useCallback(() => {
+    if (findButtonRef.current) {
+      const { offsetWidth } = findButtonRef.current;
+      if (offsetWidth === 0) {
+        prevButtonWidth.current = offsetWidth;
+        controlRef.current!.style.marginRight = '';
+        iconsContainerRef.current!.style.transform = '';
+        return;
+      }
+
+      controlRef.current!.style.marginRight = `${isValuePresent ? offsetWidth : 0}px`;
+      iconsContainerRef.current!.style.transform = `translateX(${
+        isValuePresent ? -offsetWidth : 0
+      }px)`;
+
+      iconsContainerRef.current!.style.transition =
+        prevButtonWidth.current === 0 && offsetWidth !== 0 ? 'none' : '';
+      prevButtonWidth.current = offsetWidth;
+    }
+  }, [isValuePresent]);
+
+  useIsomorphicLayoutEffect(() => {
+    recalculateComponentStyles();
+  }, [recalculateComponentStyles]);
+
+  useGlobalEventListener(window, 'resize', () => {
+    recalculateComponentStyles();
+  });
```

Comment:
а что если так? чтобы не создавать лишние функции

```suggestion
  useIsomorphicLayoutEffect(recalculateComponentStyles, [recalculateComponentStyles]);

  useGlobalEventListener(window, 'resize', recalculateComponentStyles);
```